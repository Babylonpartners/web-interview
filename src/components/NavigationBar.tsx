import * as React from 'react'
import { headerStyles } from '../styles/styles';

interface Props {
    logo: any
}

export class NavigationBar extends React.PureComponent<Props> {
    render() {
        return (
            <>
            <div 
                data-testid="navigationBarComponent"  
                className="app-header">
                <img 
                   data-testid="babylonLogo"  
                   src={this.props.logo.toString()} 
                   className="app-logo" 
                   alt="Babylon Health" 
                />
            </div>
            <hr 
               style={headerStyles} 
            />
            </>
        );
}}