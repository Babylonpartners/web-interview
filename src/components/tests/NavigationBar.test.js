import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { NavigationBar } from '../NavigationBar'

afterEach(cleanup)

test('<NavigationBar>', () => {
  const { queryByTestId, container } = render(<NavigationBar logo />)
  expect(container.firstChild).toMatchSnapshot()
  expect(queryByTestId('babylonLogo')).toBeTruthy()
})
