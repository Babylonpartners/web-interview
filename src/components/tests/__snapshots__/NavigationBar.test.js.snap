// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar> 1`] = `
<div
  class="app-header"
  data-testid="navigationBarComponent"
>
  <img
    alt="Babylon Health"
    class="app-logo"
    data-testid="babylonLogo"
    src="true"
  />
</div>
`;
